src/main/java/ch/wisv/areafiftylan/service/TicketServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package ch.wisv.areafiftylan.service;
+
+import ch.wisv.areafiftylan.exception.NotGoalUserException;
+import ch.wisv.areafiftylan.exception.TicketNotTransferrableException;
+import ch.wisv.areafiftylan.exception.TicketUnavailableException;
+import ch.wisv.areafiftylan.model.Ticket;
+import ch.wisv.areafiftylan.model.User;
+import ch.wisv.areafiftylan.model.util.TicketType;
+import ch.wisv.areafiftylan.service.repository.TicketRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class TicketServiceImpl implements TicketService {
+
+    @Autowired
+    private TicketRepository ticketRepository;
+
+    @Override
+    public Optional<Ticket> getTicketByKey(String key){
```

Comment:
Services should handle the Optional part, so that the Controller doesn't have to worry about nulls. You can simply throw the exception here.